src/CodeStyle/Core/Analyzers/AbstractCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    internal abstract class AbstractCodeStyleDiagnosticAnalyzer : DiagnosticAnalyzer
```

Comment:
>AbstractCodeStyleDiagnosticAnalyzer [](start = 28, length = 35)

Are we moving this file from Features layer to here or is this being cloned? If the former, a better approach might be to instead use linked files.

Reply:
It's a partial clone. Sharing the file would require several `#if` sections.

Reply:
Is `IBuiltInAnalyzer` the only thing causing it? If so, consider extracting out an `AbstractCommonCodeStyleDiagnosticAnalyzer` which both projects share and then have `AbstractCodeStyleDiagnosticAnalyzer : AbstractCommonCodeStyleDiagnosticAnalyzer, IBuiltInAnalyzer` in the Features layer as a thin wrapper.

---
In reply to: [222773811](https://github.com/dotnet/roslyn/pull/28873#discussion_r222773811) [](ancestors = 222773811)

Reply:
👍 I'll check it out and see how the pieces fall

Reply:
definitely curious about this.  if it's hte ibuiltinanalyzer bits, we could just use a partial type to pull out the parts, instead of #ifs